### PR TITLE
fix: add "package.json" subpath export

### PR DESCRIPTION
**Pull request by iceu-bb**
Zrzut ekranu 2023-02-3 o 16 46 23

Add a fix for the warning that occurred when starting React Native metro server.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,9 @@
       "types": "./lib/node/index.d.ts",
       "require": "./lib/node/index.js",
       "default": "./lib/node/index.mjs"
+    },
+    "./package.json": {
```
Comment (kettanaito) [inline package.json]: What's the difference between this and 

```json
{
  "./package.json": "./package.json"
}
```

I've seen [other libraries](https://github.com/ReactiveX/rxjs/blob/630d2b009b5ae4e8f2a62d9740738c1ec317c2d5/package.json#L67) using a direct path, so I assume that's equivalent to "default". Do you happen to know if there's a difference?

**Review by kettanaito — DISMISSED** [review]
Hey, @iceu-bb! Thanks for looking into this. The fix looks great, I had but one question regarding the usage of "default" and whether we could drop it and still have the same behavior.

File: package.json (hunk 1 above)
Comment (mattcosta7) [reply package.json]: @kettanaito yep, when just a string, it's default.  When an object is passed it's evaluated top to bottom, returning the first match!

Comment (kettanaito) [reply package.json]: @mattcosta7, your wizardry saves the realm once more! 🙇 Thanks, Matt.

**Comment by kettanaito** [comment]
Welcome to contributors, @iceu-bb! 🎉

**Comment by kettanaito** [comment]
## Released: v1.0.1 🎉

This has been released in v1.0.1!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v1.0.1)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/1.0.1)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**